### PR TITLE
Add macOS to lint plugins integration test

### PR DESCRIPTION
**Pull request by jmagman**
## Description

Add macOS plugin integration tests.  Validate output, Podfile.lock.

Swap from `device_info` to `url_launcher` and `url_launcher_macos`.

### CODE REVIEW

File: dev/devicelab/bin/tasks/plugin_lint_mac.dart [hunk 1]
```diff
@@ -111,13 +197,11 @@ Future<void> main() async {
...
 
       final String objcAppPath = path.join(tempDir.path, objcAppName);
-
-      final String swiftPluginPath = path.join(tempDir.path, swiftPluginName);
       final File objcPubspec = File(path.join(objcAppPath, 'pubspec.yaml'));
       String pubspecContent = objcPubspec.readAsStringSync();
       pubspecContent = pubspecContent.replaceFirst(
         '\ndependencies:\n',
-        '\ndependencies:\n  $objcPluginName:\n    path: $objcPluginPath\n  $swiftPluginName:\n    path: $swiftPluginPath\n  device_info:\n',
+        '\ndependencies:\n  $objcPluginName:\n    path: $objcPluginPath\n  $swiftPluginName:\n    path: $swiftPluginPath\n  url_launcher:\n  url_launcher_macos:\n',
```
Comment (cyanglaz) [inline dev/devicelab/bin/tasks/plugin_lint_mac.dart:204]: nits:
What is the significance of url_launcher here? Did we just randomly pick a plugin that has macos impl? If so, maybe we can add a comment to explain a little.

**Review by cyanglaz — APPROVED** [review]
LGTM % nits.

File: dev/devicelab/bin/tasks/plugin_lint_mac.dart (hunk 1 above)
Comment (jmagman) [reply dev/devicelab/bin/tasks/plugin_lint_mac.dart:204]: It was a random first-party plugin with a macOS implementation.  I'll add a comment.

Comment (jmagman) [reply dev/devicelab/bin/tasks/plugin_lint_mac.dart:206]: Done!

**Pull request merged**